### PR TITLE
Remove testReadMetadataWithRelationsConcurrentModifications override

### PR DESCRIPTION
**Pull request by krvikash**
Hoping timeout may be fixed by https://github.com/trinodb/trino/pull/16958

### CODE REVIEW

**Comment by krvikash** [comment]
After doing the stress test (`invocationCount` to 10) the following connector is failing.

`kudu`:
```
Caused by: java.lang.RuntimeException: org.apache.kudu.client.NonRecoverableException: table concur_table_750gcaux6v was deleted: Table deleted at 2023-04-13 18:04:42 UTC
```
`phoenix5`:
```
Error:  io.trino.plugin.phoenix5.TestPhoenixConnectorTest.testReadMetadataWithRelationsConcurrentModifications  Time elapsed: 4.562 s  <<< FAILURE!
com.google.common.base.VerifyException: Task did not completed before timeout; completed tasks: 0, current poll timeout: 145 s
```
`sqlserver`:
```
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Transaction (Process ID 53) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction.
```

CC: @findepi | @findinpath

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectorTest.java [hunk 1]
```diff
@@ -140,14 +140,6 @@ public void testDeleteWithLike()
                 .hasStackTraceContaining("TrinoException: " + MODIFYING_ROWS_MESSAGE);
     }
 
-    @Override
-    public void testReadMetadataWithRelationsConcurrentModifications()
-    {
-        // Under concurrently, H2 sometimes returns null table name in DatabaseMetaData.getTables's ResultSet
```
Comment (findepi) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectorTest.java:146]: keep it

**Comment by findepi** [comment]
> Hoping timeout may be fixed by #16958

for iceberg yet, not for others

**Comment by krvikash** [comment]
> > Hoping timeout may be fixed by #16958
> 
> for iceberg yet, not for others

There is no override for the iceberg. Closing this.

**Pull request closed without merging**